lib/dialects/postgres/query/pg-querycompiler.js

```diff
@@ -57,9 +57,44 @@ class QueryCompiler_PG extends QueryCompiler {
     };
   }
 
-  // Compiles an `update` query, allowing for a return value.
+  using() {
+    const usingTables = this.single.using;
+    if (!usingTables) return;
+    let sql = 'using ';
+    if (Array.isArray(usingTables)) {
+      sql += usingTables
+        .map((table) => {
+          return this.formatter.wrap(table);
+        })
+        .join(',');
+    } else {
+      sql += this.formatter.wrap(usingTables);
+    }
+    return sql;
+  }
+
+  // Compiles an `delete` query, allowing for a return value.
   del() {
-    const sql = super.del(...arguments);
+    // Make sure tableName is processed by the formatter first.
+    const { tableName } = this;
+    const withSQL = this.with();
+    const wheres = this.where();
+    // Joins in PG with delete it's not valid, we throw an error and ask to use 'using'
+    if (this.join().length > 0) {
```

Comment:
converting joins to using transparently is not feasible?

Reply:
You mean use the joins logic by add 'using' joins types with the 'using()' syntax ?
I think about that but with that implementation I cant throw an error here.
Also we have a join with using syntax inside and I quite afraid about some misunderstand.

Reply:
I'm still on the fence about this, TBH. Considering that many multi-db systems use knex specifically to abstract away db differences, transforming joins to `using` here would allow them to avoid having some conditional logic. Is it possible to always transform `join` into `using`, or you see certain case where that would fail and user would be forced to use `using` explicitly anyway?

Reply:
I see the argument. I start on 'using' function because of an argument in the original PR about avoid silent converting operations. But in fact, we always do "silent" operations in Knex, especially with oracle, when converting limit/offset functions...
I will rework it.

Reply:
I wonder if we should also keep `using` syntax for those who want to be explicit.

Reply:
I push with implicit joins to 'using' + 'where's syntax and I also keep the actual using syntax too.